src/scanner.c

```diff
@@ -218,3 +218,24 @@ scanner_scan(ScannerCtx *ctx)
 
 	return ictx.tinfo.count;
 }
+
+bool
+scanner_scan_one(ScannerCtx *ctx, bool fail_if_not_found, char *item_type)
+{
+	int num_found = scanner_scan(ctx);
+	ctx->limit = 2;
```

Comment:
What's this side effect for? Seems pretty subtle to change this limit here. Why is it needed?

Reply:
This function is supposed to make sure there is only one item to be found and otherwise error. That's why the limit makes sense